docs/references/api/resource_embedding.rst

```diff
@@ -318,10 +320,294 @@ Computed relationships have good performance as their intended design enable `in
+
+  .. code-tab:: postgresql SQL
+
+    create table addresses (
+      id int primary key generated always as identity,
+      name text,
+      city text,
+      state text,
+      postal_code char(5)
+    );
+
+    create table orders (
+      id int primary key generated always as identity,
+      name text,
+      billing_address_id int references addresses(id),
+      shipping_address_id int references addresses(id)
+    );
+
+To successfully embed ``orders`` with ``addresses``, you need to create computed relationships for the foreign keys columns you want to use:
+
+.. code-block:: postgresql
+
+  create function billing_address(orders) returns setof addresses rows 1 as $$
+    select * from addresses where id = $1.billing_address_id
+  $$ stable language sql;
+
+  create function shipping_address(orders) returns setof addresses rows 1 as $$
+    select * from addresses where id = $1.shipping_address_id
+  $$ stable language sql;
+
+Now we can unambiguously embed the billing address by specifying the ``billing_address`` computed relationship.
+
+.. tabs::
+
+  .. code-tab:: http
+
+    GET /orders?select=name,billing_address(name) HTTP/1.1
+
+  .. code-tab:: bash Curl
+
+    curl "http://localhost:3000/orders?select=name,billing_address(name)"
+
+.. code-block:: json
+
+   [
+    {
+     "name": "Personal Water Filter",
+     "billing_address": {
+       "name": "32 Glenlake Dr.Dearborn, MI 48124"
+     }
+    }
+   ]
+
+.. _recursive_o2o_embed:
+
+Recursive One-To-One
+--------------------
+
+.. tabs::
+
+  .. group-tab:: ERD
+
+    .. image:: ../../_static/presidents.png
+
+  .. code-tab:: postgresql SQL
+
+    create table presidents (
+      id int primary key generated always as identity,
+      first_name text,
+      last_name text,
+      predecessor_id int references presidents(id) unique
+    );
+
+To get either side of the Recursive One-To-One relationship, create the functions:
+
+.. code-block:: postgresql
+
+  create or replace function predecessor(presidents) returns setof presidents rows 1 as $$
+    select * from presidents where id = $1.predecessor_id
+  $$ stable language sql;
+
+  create or replace function successor(presidents) returns setof presidents rows 1 as $$
+    select * from presidents where predecessor_id = $1.id
+  $$ stable language sql;
+
+Now, to query a president with its predecessor:
+
+.. tabs::
+
+  .. code-tab:: http
+
+    GET /presidents?select=last_name,predecessor(last_name)&id=eq.2 HTTP/1.1
+
+  .. code-tab:: bash Curl
+
+    curl "http://localhost:3000/presidents?select=last_name,predecessor(last_name)&id=eq.2"
+
+.. code-block:: json
+
+  [
+    {
+      "last_name": "Adams",
+      "predecessor": {
+        "last_name": "Washington"
+      }
+    }
+  ]
+
+.. _recursive_o2m_embed:
+
+Recursive One-To-Many
+---------------------
+
+This case works the same way as a normal :ref:`Many-To-One relationship<many-to-one>`.
+It is the default behavior for when you have a table like this one:
```

Comment:
This works because we have a disambiguation hack for preferring the O2M:

https://github.com/PostgREST/postgrest/blob/28d5278d624ae262fc78345560e269be2ef99777/src/PostgREST/Plan.hs#L473-L486

As stated on https://github.com/PostgREST/postgrest/issues/2863, we plan to remove this completely. So we shouldn't rely on this behavior. Here we should define a computed rel.